### PR TITLE
Safe creation

### PR DESCRIPTION
**Pull request by antomor**
Remove the deployment of the proxy factory and tests improvement

### CODE REVIEW

**Comment by antomor** [comment]
I still have some doubts about what I could check upon successful EthersSafe creation. Maybe we could try to execute some methods, but I am not sure it makes sense. WDYT?

File: tests/creation.test.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    it('should successfully create a safeSDK instance if the threshold is set properly', async () => {
+      const ethersSafeFactory = new EthersSafeFactory(user1, {
+        safeSingletonAddress,
+        proxyFactoryAddress
+      })
+      const safeSdk = await ethersSafeFactory.createSafe({
+        owners,
+        threshold: 2
+      })
+      chai.expect(safeSdk).not.to.be.undefined
```
Comment (ilanolkies) [inline tests/creation.test.ts]: We can check for example
```suggestion
      chai.expect(await safeSdk.getOwners()).to.be.eq(owners)
      chai.expect(await safeSdk.getThreshold()).to.be.eq(2)
```

File: src/EthersSafeFactory.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  paymentToken?: string
+  payment?: number
+  paymentReceiver?: string
+}
+
+class EthersSafeFactory {
+  #safeProxyConfiguration!: SafeProxyFactoryConfiguration
+  #signer!: Signer
+
+  constructor(signer: Signer, safeProxyConfiguration: SafeProxyFactoryConfiguration) {
```
Comment (ilanolkies) [inline src/EthersSafeFactory.ts]: I would just get `proxyFactoryAddress` and `safeSingletonAddress` on the constructor. Then, accept `data`, `nonce` and `callbackAddress` as method arguments

File: src/EthersSafeFactory.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  callbackAddress?: string
+}
+
+export type SafeProxyFactoryConfiguration =
+  | SafeProxyFactoryConfigurationSimple
+  | SafeProxyFactoryConfigurationExtended
+
+interface SafeAccountConfiguration {
+  owners: string[]
+  threshold?: number
```
Comment (ilanolkies) [inline src/EthersSafeFactory.ts]: I would put this as required param (based on what you must choose in the UI outside _advanced_)

File: src/EthersSafeFactory.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+class EthersSafeFactory {
+  #safeProxyConfiguration!: SafeProxyFactoryConfiguration
+  #signer!: Signer
+
+  constructor(signer: Signer, safeProxyConfiguration: SafeProxyFactoryConfiguration) {
+    this.#safeProxyConfiguration = safeProxyConfiguration
+    this.#signer = signer
+  }
+
+  async createSafe(safeAccountConfiguration: SafeAccountConfiguration): Promise<Safe> {
```
Comment (ilanolkies) [inline src/EthersSafeFactory.ts]:
```suggestion
  async createSafe(safeAccountConfiguration: SafeAccountConfiguration, deploymentOptions: {
    data: string,
    nonce?: number,
    callbackAddress?: string
  }): Promise<Safe> {
```

**Review by ilanolkies — CHANGES_REQUESTED** [review]

**Review by ilanolkies — APPROVED** [review]

**Pull request merged**